docs/source/notes/cuda.rst

```diff
@@ -430,6 +430,52 @@ Available options:
 .. _CUDA's built-in asynchronous allocator:
     https://developer.nvidia.com/blog/using-cuda-stream-ordered-memory-allocator-part-1/
 
+.. _cuda-memory-custom-allocator:
+
+Using custom memory allocators for CUDA
+---------------------------------------
+
+It is possible to define allocators as simple functions in C/C++ and compile
+them as a shared library, the code below shows a basic allocator that just
+traces all the memory operations.
+
+.. code:: C++
+   #include <sys/types.h>
+   #include <cuda_runtime_api.h>
+   #include <iostream>
+   // Compile with nvcc alloc.cc -o alloc.so -shared --compiler-options '-fPIC 
```

Comment:
Or with `g++ alloc.cc -o alloc.so -I/usr/local/cuda/include -shared -fPIC` (There's no CUDA device code in this example, so NVCC is not required)

Also, missing a ':

```suggestion
   // Compile with nvcc alloc.cc -o alloc.so -shared --compiler-options '-fPIC'
```

Reply:
Yeah, I was too lazy to write the `-I` flag :)